packages/admin-next/dashboard/src/routes/promotions/common/edit-rules/components/rules-form-field/rules-form-field.tsx

```diff
@@ -26,54 +26,49 @@ type RulesFormFieldType = {
     | "application_method.target_rules"
 }
 
-const generateRuleAttributes = (rules?: PromotionRuleResponse[]) =>
-  (rules || []).map((rule) => ({
-    id: rule.id,
-    required: rule.required,
-    field_type: rule.field_type,
-    disguised: rule.disguised,
-    attribute: rule.attribute!,
-    operator: rule.operator!,
-    values:
-      rule.field_type === "number"
-        ? rule.values
-        : rule?.values?.map((v: { value: string }) => v.value!),
-  }))
-
 export const RulesFormField = ({
   form,
   ruleType,
   setRulesToRemove,
   rulesToRemove,
   scope = "rules",
-  promotionId,
+  promotion,
 }: RulesFormFieldType) => {
   const { t } = useTranslation()
   const formData = form.getValues()
   const { attributes } = usePromotionRuleAttributes(ruleType, formData.type)
-  const { operators } = usePromotionRuleOperators()
 
   const { fields, append, remove, update, replace } = useFieldArray({
     control: form.control,
     name: scope,
     keyName: scope,
   })
 
-  const promotionType: string = useWatch({
-    control: form.control,
-    name: "type",
-  })
+  const promotionType: string =
+    useWatch({
+      control: form.control,
+      name: "type",
+    }) || promotion?.type
```

Comment:
Q: why dont we use default value in the useWatch?

Reply:
because I didn't know we could do that 😄 nice!

Reply:
ahah no problem, I thought I might be missing a behavior 😄